### PR TITLE
add Pagination/Menu wrapper

### PR DESCRIPTION
**Pull request by agoelzer**
- the Menu wrapper simplifies the component by combining the action button with the popup menu. The action button is now a child of the "Menu" component and the menu items are now part of the "items" property. If there are no child elements, it will be a regular menu (no popup).
- prepended classnames with "Nuo" to distinguish from non-NuoDBaaS components
-

### CODE REVIEW

**Review by adriansuarez — APPROVED** [review]
Looks good.

File: ui/src/components/controls/Pagination.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    "data-testid"?: string,
+    count: number,
+    page: number,
+    setPage: (page: number) => void
+}
+export default function Pagination(props: PaginationProps): JSX.Element {
+    const { count, page, setPage } = props;
+
+    if (isMaterial()) {
+        return <MuiStack spacing={2} style={{ alignItems: "center" }}>
```
Comment (sivanov-nuodb) [inline ui/src/components/controls/Pagination.tsx:16]: Should the pagination be displayed if there are no items?
It looks strange to have an empty page list.

**Review by sivanov-nuodb — APPROVED** [review]
Looks good.

File: ui/src/components/controls/Pagination.tsx (hunk 1 above)
Comment (agoelzer) [reply ui/src/components/controls/Pagination.tsx:16]: good point - fixed - it's no longer showing if there are 0 items.

File: ui/src/components/pages/ListResource.tsx [hunk 2]
```diff
@@ -96,14 +95,9 @@ export default function ListResource({ schema }: SchemaType) {
 
     function renderPaging() {
         const lastPage = Math.ceil(allItems.length / pageSize);
-        if (lastPage <= 1) {
-            return null;
-        }
-        return <Stack spacing={2} style={{ alignItems: "center" }}>
-            <Pagination count={lastPage} page={page} onChange={(event, page) => {
+        return <Pagination count={lastPage} page={page} setPage={(page) => {
```
[the remark this answers is not in the document]
Comment (agoelzer) [reply ui/src/components/pages/ListResource.tsx:98]: currently we are defaulting to 20 items per page (see ListResource.tsx "pageSize". Let's address this in another PR where we implement a page size selector.

File: ui/src/components/controls/Menu.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                    item.onClick();
+                }}>{item.label}</MenuItem>)}
+            </MuiMenu>;
+        }
+        const rect = anchor?.getBoundingClientRect();
+        const x = align === "right" ? rect?.right : rect?.left;
+        return <div style={{ display: anchor ? "block" : "none", position: "fixed", right: 0, left: 0, top: 0, bottom: 0, backgroundColor: "transparent" }} onClick={() => setAnchor(null)}>
+            <div style={{ position: "absolute", right: x, left: x, top: rect?.bottom, bottom: rect?.bottom }}>
+                <ul id="NuoMenuPopup" className={"NuoMenuPopup " + (align === "right" ? " NuoAlignRight" : " NuoAlignLeft")}>
+                    {items.map(item => <li key={item.id} className="NuoMenuPopupItem" onClick={() => item.onClick()}>{item.label}</li>)}
```
[the remark this answers is not in the document]
Comment (agoelzer) [reply ui/src/components/controls/Menu.tsx:58]: fixed

**Pull request merged**